### PR TITLE
Testing

### LINKED ISSUE #21: Implement backup and restore
Now that the app can be installed, we should implement backup and restore scripts.

### LINKED ISSUE #22: Implement upgrade mechanism
Now that the app can be installed, we should allow our users to upgrade the app (especially because we have no backup script yet 😅).
When backup&restore tools will be implemented, we should implement a backup-before-upgrade tool.

### LINKED ISSUE #23: Allow to change URL
We need to implement change_url script. https://yunohost.org/#/packaging_apps_scripts

### PR DESCRIPTION
**Pull request by lapineige**
## Problem

- Fix #21 
- Fix #22
- Fix #23

## Solution

- Implement `backup` and `restore` scripts [#21](https://github.com/YunoHost-Apps/codimd_ynh/issues/21)
- Implement `upgrade` script [#22](https://github.com/YunoHost-Apps/codimd_ynh/issues/22)
- Implement `change_url` script [#23](https://github.com/YunoHost-Apps/codimd_ynh/issues/23).
- improve `README`
- Add `README_fr`
- Add LDAP authentication
- Merge #28 

## PR Status
- [x] Code finished.
- [X] Tested with Package_check.
- [x] Fix or enhancement tested.
- [X] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Package_check results
---
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/codimd_ynh%20PR31%20(yalh76)/)

### CODE REVIEW

File: README_fr.md
Comment (SISheogorath) [inline README_fr.md]: Sorry, but this will only cause confusion. We don't host any documentation on HackMD.io. (This documentation is for the HackMD team version of CodiMD and unreleated to the fork)

As above, please use https://github.com/codimd/server/tree/master/docs/

Or point it to a version specific documentation (e.g. https://github.com/codimd/server/tree/1.6.0/docs/) :)

File: README.md
Comment (SISheogorath) [inline README.md]: Same as above, this will point to the wrong documentation.

File: conf/config.json.example
Comment (SISheogorath) [inline conf/config.json.example]: Is it intended to actually use free-URL-mode, while it provides a etherpad-like experience, I wouldn't recommend it, as it sometimes causes issues with unintended pathnames that are used, for example, by bots that end up as notes (e.g. `wp-login.php` suddenly becomes a note because some odd bot thought it would be great to scan for wordpress)

**Review by SISheogorath — COMMENTED** [review]
:wave: CodiMD maintainer here :slightly_smiling_face: 

Some hints about confusion that can appear due to CodiMD existing in two different ways (we are working on resolving this situation) but generally speaking, love it!

I don't know if you are already around in [our community chat](https://riot.im/app/#/room/#codimd:matrix.org) but if not, feel free to say hi :)

Also once this is merge and going for production deployments feel free to open a PR at our install guides:

https://github.com/codimd/server/tree/master/docs/setup

File: README_fr.md
Comment (lapineige) [reply README_fr.md]: Ok, my bad, I probably picked the wrong documentation or didn't change it when the project was forked.

File: conf/config.json.example
Comment (ericgaspar) [reply conf/config.json.example]: Thanks very much for the review!

File: README_fr.md
Comment (yalh76) [reply README_fr.md]: documentation link changed to https://github.com/codimd/server/tree/master/docs/

File: README.md
Comment (yalh76) [reply README.md]: documentation link changed to https://github.com/codimd/server/tree/master/docs/

File: conf/config.json.example
Comment (yalh76) [reply conf/config.json.example]: @ericgaspar and @lapineige do we switch to `"allowFreeURL": false,` ?

Comment (ericgaspar) [reply conf/config.json.example]: @yalh76  I would put `allowFreeURL` to `false` for now. This option could be documented and activated as an option by users.

Comment (yalh76) [reply conf/config.json.example]: `allowFreeURL `disbaled

**Review by yalh76 — APPROVED** [review]

**Comment by yalh76** [comment]
Revision: 29129b2ddeb9fb22c7c7b562cc1e1e04b93b0c19 
### Stretch
image

### Buster
image

**Comment by ericgaspar** [comment]
Shall we merge in 3 days ? :)

**Comment by lapineige** [comment]
Go for it :)

**Comment by yalh76** [comment]
Shall we merge now ?

**Pull request merged**
